src/Common/NamedCollections/NamedCollectionsMetadataStorage.cpp

```diff
@@ -150,22 +150,22 @@ class NamedCollectionsMetadataStorage::LocalStorage : public INamedCollectionsSt
             out.sync();
         out.close();
 
-        fs::rename(tmp_path, getPath(path));
+        fs::rename(tmp_path, getPath(file_name));
     }
 
-    void remove(const std::string & path) override
+    void remove(const std::string & file_name) override
     {
-        if (!removeIfExists(getPath(path)))
+        if (!removeIfExists(file_name))
```

Comment:
This is the bugfix: `getPath` is called also in `removeIfExists`, so no need to call that here.